consensus/ssz_types/Cargo.toml

```diff
@@ -19,3 +19,6 @@ arbitrary = { version = "0.4.4", features = ["derive"], optional = true }
 [dev-dependencies]
 serde_yaml = "0.8.11"
 tree_hash_derive = "0.2.0"
+
+[features]
+with-arbitrary = ["arbitrary"]
```

Comment:
Features and optional dependencies share a namespace by design, so I don't think it's bad to have an optional dependency without a corresponding `with-X` feature. At least that's my interpretation of the docs on features, and what I've seen elsewhere: https://doc.rust-lang.org/cargo/reference/features.html

Great catch though with the `fake_crypto` and `libp2p-websocket` finds!

Reply:
My pref is for no `with-` prefix, but if @paulhauner doesn't mind I'll go with it

Reply:
Oh sorry, this fell off my radar.

I think my preference is without `with-`. Only because it feels like if you add the `with-` here, then if something else imported these crates then they'd need `with-with-arbitrary`. Is that right or am I misunderstanding?

Reply:
Nah there'd be no need for `with-with`, the only time you need to use a name other than arbitrary is when you have a crate that optionally depends on `arbitrary`, _and_ you want to enable features (probably `arbitrary`) of your deps when `arbitrary` is enabled. For the places where Kirk needed this I think he used `arbitrary-fuzz`, like:

https://github.com/sigp/lighthouse/blob/7818447fd20e96d2e1f9e48c1ff30d602c785738/consensus/state_processing/Cargo.toml#L34-L46

Reply:
Yes, basically the options are:
1. Revert the `with-arbitrary` feature (it wouldn't appear under the `features` in the toml.
2. Keep the `with-arbitrary` feature.
3. Find another name if you guys prefer :)

Honestly it's a matter of personal preference and @michaelsproul  is leaning towards option n1 :)

Reply:
I see, my bad.

I'm really not fussed, I would lean *ever so slightly* to without-with. Consider that Michael is also there, lets go with that.

Sorry to be a PITA @pscott! <3

Reply:
Ay np! :)